src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -22,6 +22,16 @@ def _resolve_aliases(target):
     else:
       yield declared
 
+    for export in _resolve_exports(declared):
```

Comment:
A function named `_resolve_aliases` is a slightly unexpected place to find this logic, since it has nothing to do with aliases, although it's clearly the most straightforward way to implement this. Maybe clarify by renaming `_resolve_aliases` to `_resolve_strict_dependencies`?